kolibri/utils/options.py

```diff
@@ -1,10 +1,36 @@
 import logging.config
 import os
+import sys
 
 from configobj import ConfigObj
 from configobj import flatten_errors
 from configobj import get_extra_values
 from validate import Validator
+try:
+    import kolibri.core.analytics.pskolibri as psutil
+except NotImplementedError:
+    # This module can't work on this OS
+    psutil = None
+
+
+def calculate_thread_pool():
+    """
+    Returns the default value for CherryPY thread_pool
+    It is calculated based on the best values obtained in
+    several partners installations.
+    The value must be between 10 (default CherryPy value) and 200.
+    Servers with more memory can deal with more threads.
+    Calculations are done for servers with more than 2 Gb of RAM
+    """
+    pool = 10
+    if psutil:
+        total_memory = psutil.virtual_memory().total / pow(2, 30)  # in Gb
+        if total_memory > 2:
+            pool = int(45 * total_memory - 80)  # 10 for 2 Gb,100 for 4 Gb
+            pool = 200 if pool > 200 else pool
```

Comment:
I did some local tests on my machine (only using siege on `127.0.0.1:8080`), and increasing the thread pool beyond 100 seemed to have a negative impact on performance. Is it possible that after a certain amount of threads, the context switching issues that we had previously seen (and made us reduce the default thread pool value) might start to dominate?

I wonder if we should start at 50, and then max out at 100 to be a little more conservative in this PR?

Reply:
In the tests we did in Honduras, with 8 Gb of RAM, 150 was working better than 100.